Add a WKWebPushToolMain for webpushtool to use

beidson opened the pull request:
#### a12bd8488608889a0dc31cdaed702a325b9b7d8b
<pre>
Add a WKWebPushToolMain for webpushtool to use
<a href="https://bugs.webkit.org/show_bug.cgi?id=261235">https://bugs.webkit.org/show_bug.cgi?id=261235</a>
rdar://115079480

Reviewed by Tim Horton and Andy Estes.

Upcoming webpushtool work requires its code to be part of the WebKit build, as opposed to it being a WebKit client.
So let&apos;s make it use WKMain like our daemons do.

* Source/WebKit/Configurations/WebKit.xcconfig:
* Source/WebKit/Configurations/webpushtool.xcconfig:
* Source/WebKit/Shared/API/Cocoa/WKMain.h:
* Source/WebKit/Shared/API/Cocoa/WKMain.mm:
(WKWebPushToolMain):
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/webpushd/webpushtool/WebPushToolMain.h: Copied from Source/WebKit/Shared/API/Cocoa/WKMain.h.
* Source/WebKit/webpushd/webpushtool/WebPushToolMain.mm:
(WebKit::WebPushToolMain):
(main): Deleted.
* Source/WebKit/webpushd/webpushtool/webpushtool.cpp: Copied from Source/WebKit/Shared/API/Cocoa/WKMain.mm.
(main):

Canonical link: <a href="https://commits.webkit.org/267714@main">https://commits.webkit.org/267714@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/41ca64d91a422af3ad919de26188f0f3b94f4ccf

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/17493 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/17818 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/18335 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/19287 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/16363 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/21096 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/17962 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/18496 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/17701 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/18009 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/15190 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/20107 "Built successfully") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/39/builds/15242 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/22552 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/16253 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/16084 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/20395 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/16667 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/14125 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/15795 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/4163 "Built successfully and passed tests") | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/4/builds/20163 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/16508 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aestes commented on Source/WebKit/Shared/API/Cocoa/WKMain.h:
```diff
@@ -35,6 +35,7 @@ extern "C" {
 WK_EXPORT int WKXPCServiceMain(int argc, const char** argv) WK_API_AVAILABLE(macos(10.15), ios(13.0));
 WK_EXPORT int WKAdAttributionDaemonMain(int argc, const char** argv) WK_API_AVAILABLE(macos(13.0), ios(16.0));
 WK_EXPORT int WKWebPushDaemonMain(int argc, char** argv) WK_API_AVAILABLE(macos(13.0), ios(16.0));
+WK_EXPORT int WKWebPushToolMain(int argc, char** argv) WK_API_AVAILABLE(macos(14.0), ios(17.0));
```
Should this use the `_TBA` macros instead?

beidson replied:
Searching through the codebase, nope that ship has sailed.

aestes replied:
But surely at this point it's wrong to say this will be available in macOS 14 and iOS 17

beidson replied:
Yes surely!

beidson committed:
Add a WKWebPushToolMain for webpushtool to use
https://bugs.webkit.org/show_bug.cgi?id=261235
rdar://115079480

Reviewed by Tim Horton and Andy Estes.

Upcoming webpushtool work requires its code to be part of the WebKit build, as opposed to it being a WebKit client.
So let's make it use WKMain like our daemons do.

* Source/WebKit/Configurations/WebKit.xcconfig:
* Source/WebKit/Configurations/webpushtool.xcconfig:
* Source/WebKit/Shared/API/Cocoa/WKMain.h:
* Source/WebKit/Shared/API/Cocoa/WKMain.mm:
(WKWebPushToolMain):
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/webpushd/webpushtool/WebPushToolMain.h: Copied from Source/WebKit/Shared/API/Cocoa/WKMain.h.
* Source/WebKit/webpushd/webpushtool/WebPushToolMain.mm:
(WebKit::WebPushToolMain):
(main): Deleted.
* Source/WebKit/webpushd/webpushtool/webpushtool.cpp: Copied from Source/WebKit/Shared/API/Cocoa/WKMain.mm.
(main):

Canonical link: https://commits.webkit.org/267714@main
